core/src/broadcast_stage.rs

```diff
@@ -28,8 +28,7 @@ pub enum BroadcastStageReturnType {
 
 struct Broadcast {
     id: Pubkey,
-
-    coding_generator: CodingGenerator,
+    session: erasure::Session,
```

Comment:
this is not going to work as-is...

CodingGenerator was saving up data blobs as broadcast went, emitting coding blobs when enough data blobs had been seen by the generator....  I see no facility in erasure::Session for saving previous data blobs.

broadcast almost always calls the generator with one blob at a time

I think you should back out the generator changes and focus on the recover stuff, which is nearly context-free.  the generator side should stay the way it is for now.